AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs

```diff
@@ -17,46 +17,32 @@ public AcademyGateway(AcademyContext academyContext)
             _academyContext = academyContext;
         }
 
-        public List<ResidentInformation> GetAllResidents(string postcode = null, string address = null)
+        public List<ResidentInformation> GetAllResidents(int cursor, int limit, string firstname = null,
+            string lastname = null, string postcode = null, string address = null)
         {
-            var addressesWithNoFilters = _academyContext.Addresses
+            var addressesFilteredByPostcode = _academyContext.Addresses
                 .Include(p => p.Person)
+                .Where(a => string.IsNullOrEmpty(address) || a.AddressLine1.ToLower().Replace(" ", "").Contains(StripString(address)))
+                .Where(a => string.IsNullOrEmpty(postcode) || a.PostCode.ToLower().Replace(" ", "").Equals(StripString(postcode)))
+                .Where(a => string.IsNullOrEmpty(firstname) || a.Person.FirstName.ToLower().Replace(" ", "").Equals(StripString(firstname)))
+                .Where(a => string.IsNullOrEmpty(lastname) || a.Person.LastName.ToLower().Replace(" ", "").Equals(StripString(lastname)))
+                .Where(a => a.Person.Id > cursor)
```

Comment:
I couldn't see any tests around cursor & limit, unless I missed them. Also is the limit being used at all?